provider/clusterrosaclassic/cluster_rosa_classic_resource.go

```diff
@@ -995,6 +1027,17 @@ func (r *ClusterRosaClassicResource) Read(ctx context.Context, request resource.
 	response.Diagnostics.Append(diags...)
 }
 
+func validateAdminCredentialsChange(ctx context.Context, state,
+	plan *ClusterRosaClassicState) diag.Diagnostics {
+	diags := diag.Diagnostics{}
+	stateUsername, statePassword := expandAdminCredentials(ctx, state.AdminCredentials, diags)
+	planUsername, planPassword := expandAdminCredentials(ctx, plan.AdminCredentials, diags)
+	if planUsername != "" && stateUsername != planUsername || planPassword != "" && statePassword != planPassword {
```

Comment:
Is it possible to create the AdminCredentials after the cluster creation?
Following this if statement I understand it possible, but is it possible by the API?
If it possible, so should the update also check if the boolean `create_admin_user` was changed from false to true?

Reply:
added the check for `create_admin_user`
also added the subsystem test
@nirarg please help to take a look, thanks